NOTES.md

```diff
@@ -0,0 +1,270 @@
+
+```
+_Frontend Request_
+_Notes_ (specificlly about the Frontend)
+_Backend Response_
+_Notes_ (specifically about the Backend)
+```
+
+## Errors
+
+_Notes_ => If there is an error in any of the functions we return =>
+
+```
+{
+  "error": {
+    "message": "The error message"
+  }
+}
+```
+
+## Users
+
+### /users/
+
+- Add User
+  **post**
+  _Frontend Request_ =>
+
+  ```
+  {
+    "username": "username",
+    "email": "email@email.com",
+    "password": "password",
+    "address": {
+      "city": "city"
+    }
+  }
+  ```
+
+  _Notes_ => That was the minimal request required, other fields that can be added include =>
+
+  ```
+  {
+    "firstName": "firstName",
+    "lastName": "lastName",
+    "avatar": "avatar",
+    "address": {
+      "country": "country"
+    }
+  }
+  ```
+
+  _Backend Response_ =>
+
+  ```
+  {
+    "avatar": "/statics/01.png",
+    "points": 0,
+    "booksToOffer": [],
+    "booksToRemember": [],
+    "booksInterestedIn": [],
+    "matches": [],
+    "_id": "60a38261cb726721a8101086",
+    "username": "username",
+    "email": "email@email.com",
+    "address": {
+      "country": "Germany",
+      "city": "city"
+    },
+    "createdAt": "2021-05-18T09:01:21.791Z",
+    "updatedAt": "2021-05-18T09:01:21.791Z"
+  }
+  ```
+
+  _Notes_ => This was based on the minimal request => This response will be refered to in future as **UserObject**
+
+### /users/:id
+
+- Get User
+  **get**
+  _Backend Response_ => **UserObject**
+
+- Update one/many Fields In A User
+  **put**
+  _Frontend Request_ => Any **UserObject** field that is not a _Mongoose ID_
+  _Notes_ => If the field is made up =>
+
+  ```
+  "This field does not exsist": "user data"
+  ```
+
+  Will **not** return an error, just the **UserObject** not changed in any way. If you try and change a field that is a _Mongoose ID_, you will recieve an error.
+  _Backend Response_ => **UserObject**
+  _Notes_ => The fields changed will be shown straight away in the response.
+
+- Delete User
+  **del**
+  _Backend Response_ => **UserObject**
+
+### /users/login
+
+- Login User
+  **post**
+  _Frontend Request_ =>
+
+  ```
+  {
+    "email": "the users email"
+  }
+  ```
+
+  _Notes_ => currently working on password implementation
+  _Backend Response_ => **UserObject**
+
+## Books
+
+### /books/
+
+- Add Book
+  **post**
+  _Frontend Request_ =>
+
+  ```
+  {
+    "title": "title",
+    "description": "description",
+    "authors": ["author"],
+    "publishedDate": Number,
+    "isbn": ["isbn"],
+    "pages": Number,
+    "shape": "shape",
+    "category": ["category"],
+    "selectedFiles": ["selected file"],
+    "owner": "a valid user mongoose id"
+  }
+  ```
+
+  _Notes_ => This is the minimal request required => The city field will default to berlin
+  _Backend Response_ =>
+
+  ```
+  {
+    "newBook": {
+      "authors": [
+        "J. K. Rowling"
+      ],
+      "isbn": [],
```

Comment:
isbn missing... could you please just throw that in?

Reply:
yes

Reply:
done

Reply:
thx